lib/absinthe/phase/parse.ex

```diff
@@ -23,7 +23,12 @@ defmodule Absinthe.Phase.Parse do
   end
 
   def run(input, options) do
-    run(%Absinthe.Blueprint{input: input}, options)
+    telemetry = %{
+      start_time: System.system_time(),
+      start_time_mono: System.monotonic_time()
+    }
+
+    run(%Absinthe.Blueprint{input: input, source: input, telemetry: telemetry}, options)
```

Comment:
Initializing all of this stuff is tricky. For example, how do we want to handle subscription documents? We probably want to re-initialize the telemetry information every time we execute the stored subscriptions. I'm inclined to say that this should get a dedicated phase just so we can make sure to loop it into the correct place.

Plus, with document providers, parsing may happen at compile time.

Reply:
I had a separate phase earlier, but I wasn't able to include parse time, since that's wired up in the current first phase.

I went down the path of splitting up the `Parse` phase to insert a `Telemetry` phase but that got messy real quick, lots of things depend on how the `Parse` phase works - do you think I should try that anyway?

The idea was to have the pipeline start with `[Init, Telemetry, Parse, ...`

Reply:
Could we parameterize the Telemetry phase?

```
    [
      {Phase.Telemetry, :start},
      # Parse Document
      {Phase.Parse, options},
      ...
      {Phase.Telemetry, :stop}
    ]
```

Reply:
Yeah, I had a version of that earlier, but the tricky part is that `Phase.Parse` expects a `Language.Source.t()` & initializes a `Blueprint`, so before it there is nowhere to save the timestamp. I tried to split it but a lot of places (especially tests) expect `Phase.Parse` to take in a string

I could take on this refactor, I think i'd create a new first phase, something like `Phase.Init` which simply takes the `Language.Source` and inserts it into a new `Blueprint`, letting the `Phase.Parse` take it from there...

I'd probably try that as a separate PR then rebase this on top of that (just to keep things from getting out of hand here)

Reply:
@benwilson512 Is this something that one needs to opt-in or is this on by default?  Would it make sense to move this to another hex package within `absinthe-graphql` organization?

Reply:
The goal of the `telemetry` project is to build instrumentation into the libraries themselves.. projects like Plug, Ecto & Phoenix are all building this into the core packages.

If there are no `telemetry` handlers installed, these will effectively be no-op calls, doing at most a fast handler lookup...